src/App.jsx

```diff
@@ -14,33 +16,106 @@ export const goodsFromServer = [
-      </button>
+function getUpdatedList(statusOfCurrentList, isReversed) {
+  const goods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  // eslint-disable-next-line max-len, prettier/prettier
+  const rev = whatReverse => (isReversed ? whatReverse.reverse() : whatReverse);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  switch (statusOfCurrentList) {
+    case SORT_ALPHABETICALLY:
+      goods.sort(
+        (good1, good2) => good1.localeCompare(good2),
+        // eslint-disable-next-line function-paren-newline
+      );
+
+      return rev(goods);
+    case SORT_BY_LENGTH:
+      goods.sort((good1, good2) => good1.length - good2.length);
+
+      return rev(goods);
+    case NOT_SORTED:
+      return rev(goods);
+    default:
+      return goodsFromServer;
+  }
+}
+
+export const App = () => {
+  const [statusOfCurrentList, setStatusOfCurrentList] = useState(NOT_SORTED);
+  const [isReversed, setIsReversed] = useState(false);
+  const showReset = statusOfCurrentList !== NOT_SORTED || isReversed;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn({
+            button: true,
+            'is-info': true,
+            'is-light': statusOfCurrentList !== SORT_ALPHABETICALLY,
+          })}
+          onClick={() => {
+            setStatusOfCurrentList(SORT_ALPHABETICALLY);
+          }}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn({
+            button: true,
+            'is-success': true,
+            'is-light': statusOfCurrentList !== SORT_BY_LENGTH,
+          })}
+          onClick={() => {
+            setStatusOfCurrentList(SORT_BY_LENGTH);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn({
+            button: true,
+            'is-warning': true,
+            'is-light': !isReversed,
+          })}
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
+        >
+          Reverse
+        </button>
+
+        {showReset && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setStatusOfCurrentList(NOT_SORTED);
+              setIsReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {getUpdatedList(statusOfCurrentList, isReversed).map(good => (
```

Comment:
1. move to a variable
2. since in a future goodsFromServer will not be static you'd better pass this value to a function;
```
const visibleGood = getUpdatedList(goodsFromServer, statusOfCurrentList, isReversed)
``` 
```suggestion
        {visibleGood.map(good => (
```